fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -214,7 +214,12 @@ private void processSimpleType(Schema schema, Schema readerSchema, JBlock method
         processEnum(readerSchema, methodBody, action, putExpressionIntoParent);
         break;
       case FIXED:
-        processFixed(schema, methodBody, action, putExpressionIntoParent, reuseSupplier);
+        // to preserve reader fixed specific options use reader field schema
+        if (action.getShouldRead() && readerSchema != null && Schema.Type.FIXED.equals(readerSchema.getType())) {
+          processFixed(readerSchema, methodBody, action, putExpressionIntoParent, reuseSupplier);
+        } else {
+          processFixed(schema, methodBody, action, putExpressionIntoParent, reuseSupplier);
+        }
```

Comment:
Nitpick: since everything is the same except the first param of the method call, we might want instead to declare a variable and determine its value based on the condition, then pass this into a single call to the method, e.g.:

```
Schema fixedFieldSchema;
if (action.getShouldRead() && readerSchema != null && Schema.Type.FIXED.equals(readerSchema.getType())) {
  // to preserve reader fixed specific options use reader field schema
  fixedFieldSchema = readerSchema;
} else {
  fixedFieldSchema = schema;
}
processFixed(fixedFieldSchema, methodBody, action, putExpressionIntoParent, reuseSupplier);
```

It is a bit more verbose, but less repetitive, and makes it extra clear what exactly is the condition affecting. Since the generator code is extremely tedious to understand, this might help readability, but feel free to disregard if you feel otherwise.